libs/mutantschemata/cpp_source/rewrite.hpp

```diff
@@ -0,0 +1,794 @@
+                    newExpr = "false: ";
+                    break;
+                default:
+                    // continue for loop
+                    continue;
+            }
+            // insert the singleton
+            createAndStoreActualMutant(binOp, newExpr, endBracket);
+        }
+        
+    }
+    
+    void createAndStoreActualMutant(const clang::BinaryOperator *binOp, const std::string &newExpr, const std::string &endBracket) {
+        
+        
+        
+        // calculate offset and FileEntry (we don't use FileID as this can change depending on the order of opening files etc.)
+        const clang::FileEntry *fE;
+        unsigned exprOffs, bracketsOffs;
+        calculateOffsetLoc(binOp->getLocStart(), fE, exprOffs);
+        calculateOffsetLoc(binOp->getLocEnd(), fE, bracketsOffs, true);
+        
+        // create and store the created (meta-) mutant
+        InsertedMutant *im = new InsertedMutant(fE, exprOffs, bracketsOffs);
+        MutantInsert *mi = new MutantInsert(im, newExpr, endBracket);
+        
+        insertedMutants.value_comp() = insertedMutants.key_comp();
+        std::set<MutantInsert*, mutantLocComp>::iterator it = insertedMutants.find(mi);
+        if (it == insertedMutants.end()) {
+            mutantInserts.push_back(mi);
+            insertedMutants.insert(mi);
+            llvm::errs() << "inserted mutant in FE: " << std::to_string((long)(mi->mutantLoc->fE)) << " @offset: [" << mi->mutantLoc->exprOffs << ", " << mi->mutantLoc->bracketsOffs << "] | " << mi->expr << "\n";
+            mutant_count++; //increase count as we are sure it isn't a duplicated vallid mutant
+        } else {
+            llvm::errs() << "duplicate mutant in FE: " << std::to_string((long)(mi->mutantLoc->fE)) << " @offset: [" << mi->mutantLoc->exprOffs << ", " << mi->mutantLoc->bracketsOffs << "] | " << mi->expr << "\n";
+        }
+    }
+    
+    /**
+     * Check if we want to mutate this file.
+     * (if it's a sourceFile and we haven't yet visited it)
+     */
+    bool isfileToMutate(clang::FullSourceLoc FullLocation) {
+        const clang::FileEntry *fE = SourceManager->getFileEntryForID(SourceManager->getFileID(FullLocation));
+        if (fE) {
+            for (std::string item: SourcePaths) {
+                if (llvm::sys::fs::equivalent(item, fE->tryGetRealPathName())) {
+                    return true;
+                    //                    return VisitedSourcePaths.find(fE->tryGetRealPathName()) == VisitedSourcePaths.end();
+                }
+            }
+        }
+        return false;
+    }
+    
+    
+public:
+    explicit MutatingVisitor(clang::CompilerInstance *CI): astContext(&(CI->getASTContext())), pp(astContext->getLangOpts()) {
+        SourceManager = &astContext->getSourceManager();
+    }
+    
+    virtual bool VisitStmt(clang::Stmt *s) {
+        clang::FullSourceLoc FullLocation = astContext->getFullLoc(s->getLocStart());
+        if (!isfileToMutate(FullLocation)) {
```

Comment:
This if-statement is redundant you can just return the condition directly and take the boolean from it

Reply:
Sorry. The indention confused me. However, it seems that this function never returns false? When should it do that